File: springdoc-openapi-tests/springdoc-openapi-groovy-tests/src/test/groovy/test/org/springdoc/api/app1/SpringDocApp1Test.java
package test.org.springdoc.api.app1;

import groovy.lang.MetaClass;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.springdoc.core.utils.SpringDocUtils;
import test.org.springdoc.api.AbstractSpringDocTest;

import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.test.context.TestPropertySource;

@TestPropertySource(properties = "springdoc.enable-groovy=false")
public class SpringDocApp1Test extends AbstractSpringDocTest {
	@BeforeAll
	public static void init() {
		SpringDocUtils.getConfig().removeRequestWrapperToIgnore(MetaClass.class);
	}

	@AfterAll
	public static void clean() {
		SpringDocUtils.getConfig().addRequestWrapperToIgnore(MetaClass.class);
	}

	@SpringBootApplication
	static class SpringDocTestApp {}
}

File: springdoc-openapi-tests/springdoc-openapi-kotlin-data-rest-tests/src/test/kotlin/test/org/springdoc/api/v31/app1/SpringDocApp1Test.kt
package test.org.springdoc.api.v31.app1

import org.springframework.boot.autoconfigure.SpringBootApplication
import test.org.springdoc.api.v31.AbstractKotlinDataRestTest

/**
 * The properties an entity inherits from a Kotlin `@Embeddable` have to be described,
 * see https://github.com/springdoc/springdoc-openapi/issues/3332
 */
class SpringDocApp1Test : AbstractKotlinDataRestTest() {

	@SpringBootApplication
	class SpringDocTestApp
}
